### DISCUSSION TITLE
Can't get directus_users in filter flow

### DISCUSSION THREAD

**Discussion by luciandavidescu** [directus/directus#17388] [Archive]
### Describe the Bug

Failing to get user info right before deleting the user, with a blocking event hook in Flows
"Internal server error: Cannot read properties of undefined (reading 'primary')"
The user is still there after the attempt.

I tried with both the key field `{{$trigger.payload}}` and with a filter like this. 
```
{
    "filter": {
        "id": {
            "_eq": "{{$trigger.payload}}"
        }
    }
}
```
Either works in other contexts than filter (with the corresponding variable).

These are the log entries of the queries, both resulting in an empty field.
```
{
  "collection": "directus_users\t",
  "key": [
    "7cbe5217-0378-44f9-96a7-58f19f2df987"
  ],
  "query": null
}
```

```
{
  "collection": "directus_users\t",
  "key": null,
  "query": {
    "filter": {
      "id": {
        "_eq": [
          "7cbe5217-0378-44f9-96a7-58f19f2df987"
        ]
      }
    }
  }
}
```

### To Reproduce

![flow](https://user-images.githubusercontent.com/26859276/216545022-277b60e0-baf1-49be-b045-3d66b2ee812b.jpg)

1. Create a Flow -> event hook / filter / items.delete / directus users + read data + directus_users + id={{$trigger.payload}} (or a filter).

2. Try to delete a user

### Errors Shown

{
  "errors": [
    {
      "message": "Cannot read properties of undefined (reading 'primary')",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}

### What version of Directus are you using?

9.22.4

### What version of Node.js are you using?

19.4.0

### What database are you using?

SQLite

### What browser are you using?

Chrome

### How are you deploying Directus?

PaaS

**Comment by paescuj** [comment]
Transformed into a discussion since this feels more like a question than an actual issue.

**Reply by luciandavidescu** [reply]
Yep, it is possible that I'm missing some obvious thing badly. 
If someone may encounter the same problem, a workaround is to get data from freshly deleted users with an 'action' query in directus_revisions.

**Reply by paescuj** [reply]
Just noticed this: Shouldn't it be `{{$trigger.key}}` in the "Read Data" operation instead of `{{$trigger.payload}}`?

**Reply by luciandavidescu** [reply]
for the delete event it's payload:

```
{
  "payload": [
    "7cbe5217-0378-44f9-96a7-58f19f2df987"
  ],
  "event": "users.delete",
  "collection": "directus_users"
}
```

**Reply by luciandavidescu** [reply]
as I went back to look for this in the logs i tested once more and the same flow that returned the error now simply works as it should... I did nothing except restart the server at some point, so... not quite a bug, indeed

**Reply by paescuj** [reply]
Hmm, interesting! Thanks for the info! 👍 Also please get back to us if the problem occurs again.
